### PR TITLE
fix: wait for MTW to exist before creating KYC event channel

### PR DESCRIPTION
**Pull request by cajubelt**
### Description

This allows the user to link their bank account in the same session that they verify their phone number. 

More details on the bug:
- the `watchKycStatus` saga creates an event channel when the app is opened that listens for updates on the firebase node `inHouseLiquidity/${accountAddress}/kycStatus`, 
- `accountAddress` is assigned to the MTW address if non-null, or EOA address otherwise
- if someone verifies their phone number, the channel is not re-created until the next time they close/open the app
- so if someone tries to connect their bank account right after verifying their phone number, it does not work

### Other changes

created shared helper function for `getWalletAddress` in `web3/saga.ts` so I could largely copy it but also keep the code DRY

### Tested

tested manually (verified phone number and linked bank account without closing the app in between)
added unit tests for `getWalletAddress` and `getMTWAddress`

### How others should test

- create a new account
- verify your phone number
- WITHOUT CLOSING THE APP, go into settings and tap 'link bank account'
- go through Persona flow
- verify that step 1 updates to 'completed' (does not show 'begin' button anymore)

### Related issues

na

### Backwards compatibility

na

### CODE REVIEW

File: packages/mobile/src/web3/saga.test.ts
Comment (jophish) [inline packages/mobile/src/web3/saga.test.ts:103]: None of these tests ever reach the `take` effect in `getAddress` right? Maybe worth having a test that uses [`dispatch`](https://redux-saga-test-plan.jeremyfairbank.com/integration-testing/dispatching.html) to mock placing the value (and improving coverage?). Huge Redux noob here, so excuse me if I'm getting something wrong!

**Review by jophish — COMMENTED** [review]
Great job catching this bug, and thanks for the clear explanation! Implementation LGTM pending coverage.

**Review by lisaqyli — APPROVED** [review]
Though I don't disagree with Joe's comments on additional test for getAddress but I also think that particular unit test has relatively low value so I'm just gonna ship this. I'll leave it to you on whether to add a test or not.

File: packages/mobile/src/web3/saga.test.ts
Comment (cajubelt) [reply packages/mobile/src/web3/saga.test.ts:103]: seems reasonable, thanks for the rec

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the task using Android Latest Alfajores Build and Latest iOS Alfajores Build V:1.25.1 and updated the status in Test rails

**iOS 14 :** 
https://valoraapp.testrail.io/index.php?/tests/view/4342&group_by=cases:section_id&group_order=asc&group_id=30

**Android 11:**
https://valoraapp.testrail.io/index.php?/tests/view/4279&group_by=cases:section_id&group_order=asc&group_id=30

**Observation:** Few components are not seen on Android app as compared with iOS App and we have raised a bug for it 
[buglink](https://github.com/valora-inc/wallet/issues/1915) 

**Devices :** Google Pixel 2XL(11.0), iPhone 12(14.7.1)

**CC:** @cajubelt 

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the task using Android Latest Alfajores Build V:1.25.2 and updated the status in Test rails

**Test rail link:**

**Android 10:**
https://valoraapp.testrail.io/index.php?/tests/view/7086&group_by=cases:section_id&group_order=asc&group_id=30

**Observation:** Few components are not seen on Android app as compared with iOS App and we have raised a bug for it

[Bug link
](https://github.com/valora-inc/wallet/issues/1915
)
**Devices**: Samsung Galaxy M40(10.0)

**CC:** @cajubelt 

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified this task using Latest Test flight Alfajores Build V:1.25.2 and updated the status in Test rails

**Test rail link:**
**iOS15:** https://valoraapp.testrail.io/index.php?/tests/view/4468

**Devices:** iPhone 13 mini (15.1.1)
CC: @cajubelt 
Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the task using Android Latest Alfajores Build V:1.25.2, Test flight Alfa jores Build V1.25.2 and updated the status in Test rails


Android 8:
https://valoraapp.testrail.io/index.php?/tests/view/7149&group_by=cases:section_id&group_order=asc&group_id=30


**Observation:** Few components are not seen on Android app as compared with iOS App and we have raised a [bug](https://github.com/valora-inc/wallet/issues/1915) for it




Devices: Samsung Galaxy J5 Prime (8.0.0)
CC: @cajubelt

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, we have verified the task using Latest Android Alfajores Build V:1.25.2 and updated the status in Test rails

**Test rail link:**
**Android 9:** https://valoraapp.testrail.io/index.php?/tests/view/7473

**Observation:** Few components are not seen on Android app as compared with iOS App and we have raised a [bug](https://github.com/valora-inc/wallet/issues/1915) for it

**Devices:** Redmi Note 5 Pro (9.0)
CC: @cajubelt
Thanks..!!

**Pull request merged**
